src/azure-cli/azure/cli/command_modules/synapse/operations/artifacts.py

```diff
@@ -0,0 +1,287 @@
+
+
+def create_pipeline_run(cmd, workspace_name, pipeline_name, reference_pipeline_run_id=None, is_recovery=None,
+                        start_activity_name=None, parameters=None):
+    client = cf_synapse_pipeline(cmd.cli_ctx, workspace_name)
+    return client.create_pipeline_run(pipeline_name, reference_pipeline_run_id, is_recovery,
+                                      start_activity_name, parameters)
+
+
+# Pipline run
+def query_pipeline_runs_by_workspace(cmd, workspace_name, last_updated_after, last_updated_before,
+                                     continuation_token=None, filters=None, order_by=None):
+    client = cf_synapse_pipeline_run(cmd.cli_ctx, workspace_name)
+    return client.query_pipeline_runs_by_workspace(RunFilterParameters(last_updated_after=last_updated_after,
+                                                                       last_updated_before=last_updated_before,
+                                                                       continuation_token=continuation_token,
+                                                                       filters=filters, order_by=order_by))
+
+
+def get_pipeline_run(cmd, workspace_name, run_id):
+    client = cf_synapse_pipeline_run(cmd.cli_ctx, workspace_name)
+    return client.get_pipeline_run(run_id)
+
+
+def query_activity_runs(cmd, workspace_name, pipeline_name, run_id, last_updated_after, last_updated_before,
+                        continuation_token=None, filters=None, order_by=None):
+    client = cf_synapse_pipeline_run(cmd.cli_ctx, workspace_name)
+    return client.query_activity_runs(pipeline_name, run_id,
+                                      RunFilterParameters(last_updated_after=last_updated_after,
+                                                          last_updated_before=last_updated_before,
+                                                          continuation_token=continuation_token,
+                                                          filters=filters, order_by=order_by))
+
+
+def cancel_pipeline_run(cmd, workspace_name, run_id, is_recursive=None):
+    client = cf_synapse_pipeline_run(cmd.cli_ctx, workspace_name)
+    return client.cancel_pipeline_run(run_id, is_recursive)
+
+
+# Trigger
+def list_triggers(cmd, workspace_name):
+    client = cf_synapse_trigger(cmd.cli_ctx, workspace_name)
+    return client.get_triggers_by_workspace()
+
+
+def get_trigger(cmd, workspace_name, trigger_name):
+    client = cf_synapse_trigger(cmd.cli_ctx, workspace_name)
+    return client.get_trigger(trigger_name)
+
+
+def create_or_update_trigger(cmd, workspace_name, trigger_name, definition_file, no_wait=False):
+    client = cf_synapse_trigger(cmd.cli_ctx, workspace_name)
+    properties = Trigger.from_dict(definition_file['properties'])
+    return sdk_no_wait(no_wait, client.begin_create_or_update_trigger,
+                       trigger_name, properties, polling=True)
```

Comment:
Remove "polling=True" as sdk_no_wait() has such logic. 
Please update other places.

Reply:
If remove "polling=True", it can't poll the result. Maybe it causes by track 2 sdk?

Reply:
If users don't provide --no-wait, the default value of polling is True. No need to add it here.

---
In reply to: [506169871](https://github.com/Azure/azure-cli/pull/15296#discussion_r506169871) [](ancestors = 506169871)